### DISCUSSION TITLE
Perform a HTTP POST request inside IL2Cpp.perform?

### DISCUSSION THREAD

**Discussion by KCD3V** [vfsfitvnm/frida-il2cpp-bridge#414] [Q&A] [answered]
I want to fire a post request inside IL2Cpp.perform(() => {
});

What options have I got for that?

**Comment by bluewave41** [comment, marked as answer]
By doing a post request...?

If the dump has networking methods then use those. If you can't then check the frida-http module: https://github.com/frida/gumjs-http

**Reply by KCD3V** [reply]
Thank you @bluewave41, gumjs-http seems to have zero documentation on how you might use it for post request. I've tried axios, node-fetch, calling a curl binary on the device and even trying to initiate Java.use('java.net.URL') to open a connection, nothing has worked so far. I'll keep trying different options and hopefully I'll get lucky.

**Reply by thinhbuzz** [reply]
@KCD3V take a look at the nodejs http module docs

> Node.js compatible [http](https://nodejs.org/api/http.html) module for [Frida](https://frida.re/)

**Reply by KCD3V** [reply]
Thank you Mr.Buzz, I will give that a look too

**Comment by KCD3V** [comment]
After playing around with Java.use("java.net.URL") a little more.
I discovered previously I was using localhost in my url connection and not the network IP of my server.
So yea I'm an idiot, localhost obviously refers to the mobile device.
